Bottom: ac20605b4f3ee5e792fed76841267b62a08cb171
Top:    532bf023404f4322819d46ecf1933cbb5822237a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 17:27:02 +0100

Refresh of fix-hid-structure-initialisati

---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 9360cb7..3e49ed3 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1035,6 +1035,8 @@ HID gcode_hid;
 void
 hid_gcode_init ()
 {
+  memset (&gcode_hid, 0, sizeof (HID));
+
   gcode_hid.struct_size         = sizeof (HID);
   gcode_hid.name                = "gcode";
   gcode_hid.description         = "G-CODE export.";
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index ca87a84..7c2f461 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1519,6 +1519,8 @@ png_set_crosshair (int x, int y, int a)
 void
 hid_png_init ()
 {
+  memset (&png_hid, 0, sizeof (HID));
+
   png_hid.struct_size = sizeof (HID);
   png_hid.name        = "png";
   png_hid.description = "GIF/JPEG/PNG export.";
